src/hooks/useForumConnect.ts

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useEffect, useState } from 'react'
+
+import useAuthContext from 'decentraland-gatsby/dist/context/Auth/useAuthContext'
+import useClipboardCopy from 'decentraland-gatsby/dist/hooks/useClipboardCopy'
+import useSign from 'decentraland-gatsby/dist/hooks/useSign'
+import Time from 'decentraland-gatsby/dist/utils/date/Time'
+
+import { Governance } from '../clients/Governance'
+import { GATSBY_DISCOURSE_CONNECT_THREAD, MESSAGE_TIMEOUT_TIME } from '../entities/User/constants'
+import { DISCOURSE_API } from '../entities/User/utils'
+import { openUrl } from '../helpers'
+
+import useTimer from './useTimer'
+
+export const THREAD_URL = `${DISCOURSE_API}${
+  DISCOURSE_API.endsWith('/') ? '' : '/'
+}t/${GATSBY_DISCOURSE_CONNECT_THREAD}/10000`
+const VALIDATION_CHECK_INTERVAL = 5 * 1000 // 5 seconds
+
+const getMessage = async () => Governance.get().getValidationMessage()
+export default function useForumConnect() {
+  const [user, userState] = useAuthContext()
+  const [sign, signState] = useSign(user, userState.provider)
+  const [copied, clipboardState] = useClipboardCopy(Time.Second)
+  const [signResolveReject, setSignResolveReject] = useState<{
```

Comment:
How about giving this resolve/reject object a name? (and the state/setState too)
`SignatureResolution` makes sense?